server/src/tenant/metadata/migration-generator/migration-generator.service.ts

```diff
@@ -80,16 +52,31 @@ export class MigrationGeneratorService {
     // Update appliedAt date for each migration
     // TODO: Should be done after the migration is successful
     pendingMigrations.forEach(async (pendingMigration) => {
-      await this.setAppliedAtForMigration(workspaceId, pendingMigration);
+      await this.tenantMigrationService.setAppliedAtForMigration(
+        workspaceId,
+        pendingMigration,
+      );
     });
 
+    await queryRunner.release();
+    // We want to destroy all connections to the workspace data source and invalidate the cache
+    // so that the next request will create a new connection and get the latest entities
+    await this.dataSourceService.disconnectFromWorkspaceDataSource(workspaceId);
```

Comment:
This part was missing. As comment says, we should invalidate the workspace datasource cache after each migration to force the next connection to create a new one with updated entities